pandas/tests/sparse/test_series.py

```diff
@@ -88,6 +88,18 @@ def setup_method(self, method):
         self.ziseries2 = SparseSeries(arr, index=index, kind='integer',
                                       fill_value=0)
 
+    def test_constructor_data_input(self):
+        # see gh-16905
+        test_dict = {1: 1}
+        test_index = [0, 1, 2]
+        test_series = pd.Series(test_dict, index=test_index)
+
+        arr = SparseSeries(test_dict)
+        assert arr.count() == len(test_dict)
+
+        arr = SparseSeries(test_series, index=test_index)
+        assert arr.count() == test_series.count()
```

Comment:
I had to use assert_series_equal since assert_sp_series_equal requires two sparse series, even if check_series_type=False.
This appears to result from line 1523 in util.testing.